lib/cisco_node_utils/interface_port_channel.rb

```diff
@@ -0,0 +1,161 @@
+# December 2015, Sai Chintalapudi
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+require_relative 'interface'
+
+# Add some interface-specific constants to the Cisco namespace
+module Cisco
+  # InterfacePortChannel - node utility class for port channel config management
+  class InterfacePortChannel < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      fail TypeError unless name.is_a?(String)
+      fail ArgumentError unless name.length > 0
+      @name = name.downcase
+      fail ArgumentError unless name.start_with?('port-channel')
```

Comment:
Since you downcased it in the previous step you should use @name for this check:

`+      fail ArgumentError unless @name.start_with?('port-channel')`